### PR TITLE
Upgrade opentelemetry to 0.18.0

### PR DESCRIPTION
**Pull request by BrynCooke**
Deal with all the changes in APIs from the OtelUpgrade and other miscellaneous improvements.

1. Metrics have changed significantly. Traits have been introduced, so we can now have cleaner integration otel and expose a regular Meter even though we are aggregating under the hood.

2. We can't actually update yet as we need https://github.com/open-telemetry/opentelemetry-rust/issues/894 to land in a release. The ability to send spans to multiple exporters has been removed accidentally from JaegerCollector.

Closes https://github.com/apollographql/router/issues/1948

- [x] Update opentelemetry
- [x] Refactor metrics
- [x] Remove spaceport
- [x] Otel config errors are only logged once.
- [x] Otel traces are not sent if `tracesIgnored` is detected.

### CODE REVIEW

File: apollo-router/src/plugins/telemetry/tracing/jaeger.rs
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/tracing/jaeger.rs:110]: So we should wait for the next release before merging right ?

**Review by bnjjj — APPROVED** [review]

File: apollo-router/src/plugins/telemetry/tracing/jaeger.rs
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/tracing/jaeger.rs:110]: Yep

File: apollo-router/src/plugins/telemetry/otlp.rs
Comment (BrynCooke) [inline apollo-router/src/plugins/telemetry/otlp.rs:172]: TODO: Remove

**Comment by abernix** [comment]
Marking this as a draft because I think we're blocked until the release.

**Comment by BrynCooke** [comment]
@bazaglia The issue has been that the latest otel release broke our Jager integration and there hasn't been another release since. I just figured out a workaround though so maybe we can get this over the line.

File: apollo-router/src/plugins/telemetry/apollo.rs
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/apollo.rs:44]: If it's a default, we should not wrap it into an `Option<>`

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/apollo_exporter.rs:188]: Use constant from http crate for header name

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/apollo_exporter.rs:190]: Use constant from http crate for header name

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs
Comment (bnjjj) [inline apollo-router/src/plugins/telemetry/apollo_exporter.rs:192]: Use constant from http crate for header name

File: NEXT_CHANGELOG.md
Comment (BrynCooke) [reply NEXT_CHANGELOG.md:38]: Yep, I added the migration already, but it's still a breaking change.

File: .circleci/config.yml
Comment (bazaglia) [inline .circleci/config.yml:61]: Is this comment mentioning Jaeger a leftover from duplicating the `jaeger_version` parameter? This is the protoc needed by the [opentelemetry-proto](https://github.com/open-telemetry/opentelemetry-rust/tree/main/opentelemetry-proto) crate.

File: apollo-router/src/plugins/telemetry/apollo.rs
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/apollo.rs]: [3152c3c](https://github.com/apollographql/router/pull/1970/commits/3152c3c206c6f686bfd1ee12ed842c6e26b75af5)

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/apollo_exporter.rs]: [6a1155b](https://github.com/apollographql/router/pull/1970/commits/6a1155b6b7fed93e3b20279fa35ce5b3f92a15ee)

File: apollo-router/src/plugins/telemetry/apollo_exporter.rs
Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/apollo_exporter.rs]: [6a1155b](https://github.com/apollographql/router/pull/1970/commits/6a1155b6b7fed93e3b20279fa35ce5b3f92a15ee)

File: .circleci/config.yml
Comment (BrynCooke) [reply .circleci/config.yml:61]: Yep

Comment (BrynCooke) [reply .circleci/config.yml]: [f44320d](https://github.com/apollographql/router/pull/1970/commits/f44320d31eaa40cb1a91ad46fe31bf541cbaada0)

**Review by bnjjj — APPROVED** [review]

**Pull request merged**
